aws/middleware/middleware.go

```diff
@@ -0,0 +1,101 @@
+package middleware
+
+import (
```

Comment:
We'll probably want to take a look at how request/retry middleware are grouped in various packages.